apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/tasks/OperationTask.java

```diff
@@ -142,12 +142,13 @@ private Optional<JobState> pollJobStatus(Attributes typeAttributes) {
     long startTime = System.currentTimeMillis();
     while (!jobFinished()) {
       long elapsedTime = System.currentTimeMillis() - startTime;
-      if (elapsedTime > timeoutMs) {
+      // Cancel job if it is queued for more than queuedTimeoutMs.
+      // Otherwise, if it is running, then keep it running until scheduler cancels it.
+      if (elapsedTime > queuedTimeoutMs) {
         Optional<JobState> job = jobsClient.getState(jobId);
-        // Do not cancel job if it is still in running state after 3 hours
-        if (job.isPresent() && !job.get().equals(JobState.RUNNING)) {
+        if (job.isPresent() && job.get().equals(JobState.QUEUED)) {
           log.info(
-              "Cancelling job: {} due to timeout for {}: jobState: {}",
+              "Cancelling job: {} due to queued timeout for {}: jobState: {}",
```

Comment:
Queued job can be successful after long queued time. I have collected some data for two weeks and it seems max queued time is 75 mins and after that job started and succeeded. I think we should just skip checking status instead of cancelling the queued tasks.

Reply:
If we skip checking status, there's no gain in the throughput. Those threads will be hanging for 12 hours instead of 3 hours. Now I'm making it 10 mins.